references/cuegen/generators/provider/testdata/valid.cue

```diff
@@ -0,0 +1,100 @@
+package test
+
+#Apply: {
+	#do:       "apply"
+	#provider: "test"
+	$params: {
+		// +usage=The cluster to use
+		cluster: string
+		// +usage=The resource to get or apply
+		resource: {
+			...
+		}
+		// +usage=The options to get or apply
+		options: {
+			// +usage=The strategy of the resource
+			threeWayMergePatch: {
+				// +usage=The strategy to get or apply the resource
+				enabled: *true | bool
+				// +usage=The annotation prefix to use for the three way merge patch
+				annotationPrefix: *"resource" | string
+			}
+		}
+	}
+	$returns: {
+		...
+	}
+}
+
+#Get: {
+	#do:       "get"
+	#provider: "test"
+	$params: {
+		// +usage=The cluster to use
+		cluster: string
+		// +usage=The resource to get or apply
+		resource: {
+			...
+		}
+		// +usage=The options to get or apply
+		options: {
+			// +usage=The strategy of the resource
+			threeWayMergePatch: {
+				// +usage=The strategy to get or apply the resource
+				enabled: *true | bool
+				// +usage=The annotation prefix to use for the three way merge patch
+				annotationPrefix: *"resource" | string
+			}
+		}
+	}
+	$returns: {
+		...
+	}
+}
+
+#List: {
+	#do:       "list"
+	#provider: "test"
+	$params: {
+		// +usage=The cluster to use
+		cluster: string
+		// +usage=The filter to list the resources
+		filter?: {
+			// +usage=The namespace to list the resources
+			namespace?: string
+			// +usage=The label selector to filter the resources
+			matchingLabels?: [string]: string
+		}
+		// +usage=The resource to list
+		resource: {
+			...
+		}
+	}
+	$returns: {
+		...
+	}
+}
+
+#Patch: {
+	#do:       "patch"
+	#provider: "test"
+	$params: {
+		// +usage=The cluster to use
+		cluster: string
+		// +usage=The resource to patch
+		resource: {
+			...
+		}
+		// +usage=The patch to be applied to the resource with kubernetes patch
+		patch: {
```

Comment:
The patch data in pkg is like:

```
		// +usage=The patch to be applied to the resource with kubernetes patch
		patch: *{
			// +usage=The type of patch being provided
			type: "merge"
			data: {...}
		} | {
			// +usage=The type of patch being provided
			type: "json"
			data: [...{...}]
		} | {
			// +usage=The type of patch being provided
			type: "strategic"
			data: {...}
		}
```

This is because list and struct are different in CUE. But in GO, we can cover that with any.
Can we use tag to distinguish the differences in Go in generator?

Reply:
Or we can change the way we write in Go code.

Reply:
Perhaps a little difficult, tag's expressiveness is poor. And for go any, what is the difference between `[]{...}`  and `{...}` ?

Reply:
For go, there are no differences. But in CUE, it's different. Checkout  https://cuelang.org/play/?id=PhxG9btJpjZ#cue@export@cue